forte/data/readers/conll03_reader_new.py

```diff
@@ -0,0 +1,200 @@
+# Copyright 2019 The Forte Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import os
+from typing import Iterator, Any
+from ft.onto.base_ontology import Token, Sentence, Document, EntityMention
+from forte.data.data_pack import DataPack
+from forte.data.data_utils_io import dataset_path_iterator
+from forte.data.readers.base_reader import PackReader
+
+__all__ = [
+    "CoNLL03Reader"
+]
+
+
+class CoNLL03Reader(PackReader):
+    r""":class:`CoNLL03Reader` is designed to read in the CoNLL03 dataset.
+
+    The datatset is from the following paper,
+    Sang, Erik F., and Fien De Meulder. "Introduction to the CoNLL-2003
+    shared task: Language-independent named entity recognition."
+    arXiv preprint cs/0306050 (2003).
+
+    Data could be downloaded from
+    https://deepai.org/dataset/conll-2003-english
+
+    Data format:
+    Data files contains one line "-DOCSTART- -X- -X- O" to represent the
+    start of a document. After that, each line will contain one word and
+    an empty line represent the start of a new sentence. Each line contains
+    four fields, the word, its part-of-speech tag, its chunk tag and its
+    named entity tag.
+
+    Example:
+        EU NNP B-NP B-ORG
+        rejects VBZ B-VP O
+        German JJ B-NP B-MISC
+        call NN I-NP O
+        to TO B-VP O
+        boycott VB I-VP O
+        British JJ B-NP B-MISC
+        lamb NN I-NP O
+        . . O O
+    """
+
+    def _collect(self, conll_directory) -> Iterator[Any]:  # type: ignore
+        r"""Iterator over conll files in the data_source.
+
+        Args:
+            conll_directory: directory to the conll files.
+
+        Returns: Iterator over files in the path with conll extensions.
+        """
+        logging.info("Reading .conll from %s", conll_directory)
+        return dataset_path_iterator(conll_directory, "conll")
+
+    def _cache_key_function(self, collection: str) -> str:
+        return os.path.basename(collection)
+
+    def _parse_pack(self, collection: str) -> Iterator[DataPack]:
+        doc = open(collection, "r", encoding="utf8")
```

Comment:
Maybe we should replace doc open close with `with open(collection) as doc`.

Reply:
Got it.